Notes

UkrainianBanderasCat opened the pull request:
This PR adds notes, that can be edited. Still in progress

<img width="874" alt="Screenshot 2025-04-17 at 20 23 08" src="https://github.com/user-attachments/assets/6e2b1bc5-d0ce-46ca-aa2a-4f5f07ac82fb" />

UPD:

Made new design. Notes color can now be chosen and notes text can now be edited.
This was requested in https://github.com/SebLague/Digital-Logic-Sim/issues/282 and also other users of Digital Logic Sim 
<img width="1278" alt="Screenshot 2025-04-18 at 16 54 41" src="https://github.com/user-attachments/assets/02bc817b-e346-4cc9-9235-6fbd69f7ad20" />
<img width="602" alt="Screenshot 2025-04-18 at 16 57 53" src="https://github.com/user-attachments/assets/a7ba790e-58b4-4ba3-99a3-313e63e4d5bf" />

UkrainianBanderasCat committed:
added HEX display mode to ROM and pins

UkrainianBanderasCat committed:
Added notes instance

UkrainianBanderasCat committed:
Merge branch 'notes' of https://github.com/Lime-Game-Modding/Digital-Logic-Sim-Fan-Edit into notes

UkrainianBanderasCat committed:
Removed HEX display

UkrainianBanderasCat committed:
Added note text poppup

UkrainianBanderasCat committed:
Added textArea

UkrainianBanderasCat committed:
Added different colors to notes

firecerne committed:
Merge branch 'main' into notes

UkrainianBanderasCat committed:
Added note resize

UkrainianBanderasCat committed:
Merge branch 'notes' of https://github.com/Lime-Game-Modding/Digital-Logic-Sim-Fan-Edit into notes

UkrainianBanderasCat committed:
Small fixes

UkrainianBanderasCat committed:
Added note saving

UkrainianBanderasCat committed:
Fixed note delete function

UkrainianBanderasCat committed:
Some improvements

UkrainianBanderasCat committed:
Some fixes

UkrainianBanderasCat committed:
Note color is random on creation

UkrainianBanderasCat committed:
Added shortcut

UkrainianBanderasCat committed:
Removed temp debugging

UkrainianBanderasCat committed:
Fixed bug with textarea

UkrainianBanderasCat committed:
Smol fix

SebLague commented:
This is looking very cool, but my crude text system needs some work before it can handle this multi-line stuff properly. Mouse selection, arrow keys, ctrl+A to select all, etc. all don't function correctly here. Line wrap is also not well-behaved (splits words, and adds space on new line unnecessarily)

UkrainianBanderasCat commented:
> This is looking very cool, but my crude text system needs some work before it can handle this multi-line stuff properly. Mouse selection, arrow keys, ctrl+A to select all, etc. all don't function correctly here. Line wrap is also not well-behaved (splits words, and adds space on new line unnecessarily)

So what to do now? I can continue working on fixing this problems and improving text area

SebLague commented:
You're welcome to dive into the text system and fix these issues yourself if you wish to, otherwise you can wait until I have time to do it (but might take a while).

UkrainianBanderasCat commented:
> You're welcome to dive into the text system and fix these issues yourself if you wish to, otherwise you can wait until I have time to do it (but might take a while).

Ok, I will take try, and thank you for feedback:)

UkrainianBanderasCat committed:
big textarea improvements

UkrainianBanderasCat committed:
selection not loosing focus bug fix

UkrainianBanderasCat committed:
fixed really anoying bug with selecting

UkrainianBanderasCat committed:
fixed newline bug

UkrainianBanderasCat committed:
Added limitation

UkrainianBanderasCat committed:
Merge branch 'main' into notes

UkrainianBanderasCat commented:
> This is looking very cool, but my crude text system needs some work before it can handle this multi-line stuff properly. Mouse selection, arrow keys, ctrl+A to select all, etc. all don't function correctly here. Line wrap is also not well-behaved (splits words, and adds space on new line unnecessarily)

Hello, I fixed this errors.

UkrainianBanderasCat committed:
Delete TestData directory

UkrainianBanderasCat committed:
Revert "Delete TestData directory"

This reverts commit 627ca20d2420879b8d03b02721869e14c8367084.

UkrainianBanderasCat committed:
Revert "Merge branch 'main' into notes"

This reverts commit 8bf0f51407b231f565d8b5a9ece13ff6148b38d6, reversing
changes made to 3a008dcf34c1dc8471a5de76ddccea0d1e4082cf.

UkrainianBanderasCat committed:
removed debug

SebLague commented:
Hi, I see you reverted the main branch merge -- is this still in progress?

UkrainianBanderasCat committed:
Delete TestData/Deleted Projects/MainTest/Chips directory

UkrainianBanderasCat commented:
> Hi, I see you reverted the main branch merge -- is this still in progress?

Hi, sorry for mess of commits, there are some problems with my commits, I am fixing them right now. But notes are already finished.

UkrainianBanderasCat committed:
Reapply "Merge branch 'main' into notes"

This reverts commit 94d2b2d06000c9e911b2051b6a6320528e1d6923.

UkrainianBanderasCat committed:
Fixed

UkrainianBanderasCat commented:
> Hi, I see you reverted the main branch merge -- is this still in progress?

Ok, now is all fixed and notes are finished. I fixed the problems with text area.

SebLague commented:
Hmm I'm getting some compiler errors when opening the pr

UkrainianBanderasCat commented:
> Hmm I'm getting some compiler errors when opening the pr

Hmm, that's interesting. I will figure it out

logic-mindful committed:
Fixed compile errors

UkrainianBanderasCat committed:
Update UI.cs

UkrainianBanderasCat commented:
> Hmm I'm getting some compiler errors when opening the pr

Can you try again? I think the errors are now fixed.

SebLague commented:
Okay doing a quick test -- it's looking good overall, but noticed some bugs:
* When you edit a new note, the text from prev note is still there
* Selecting text and pressing delete can cause arg out of range exception
* Selecting text and pressing arrow keys doesn't jump to start end as expected
* Jump to next/prev word with ctrl + arrow doesn't work properly (when prev/next char is a space I think)
* Clicking on text with mouse seems to work okay in first few lines, but gets progressively worse the further down you go.
* When you have run-on lines, entering text near the end of the line then causes caret to jump to wrong place (also applies when pasting text)
* Caret blink whiles typing, blink timer should be reset in this case
* Moving a note works with the undo/redo system, but adding/deleting does not

Finally, more of a design thing -- I'm not sure about the notes expanding out horizontally until you do manual new line. I think it should be fixed width and follow the automatic line breaks present in the editing window.

UkrainianBanderasCat commented:
> Okay doing a quick test -- it's looking good overall, but noticed some bugs:
> 
> * When you edit a new note, the text from prev note is still there
> * Selecting text and pressing delete can cause arg out of range exception
> * Selecting text and pressing arrow keys doesn't jump to start end as expected
> * Jump to next/prev word with ctrl + arrow doesn't work properly (when prev/next char is a space I think)
> * Clicking on text with mouse seems to work okay in first few lines, but gets progressively worse the further down you go.
> * When you have run-on lines, entering text near the end of the line then causes caret to jump to wrong place (also applies when pasting text)
> * Caret blink whiles typing, blink timer should be reset in this case
> * Moving a note works with the undo/redo system, but adding/deleting does not
> 
> Finally, more of a design thing -- I'm not sure about the notes expanding out horizontally until you do manual new line. I think it should be fixed width and follow the automatic line breaks present in the editing window.

Thanks for your feedback) I will fix this problems as soon as possible.

SebLague commented:
Sure thing! There may be more issues -- I didn't test too extensively, so maybe try a bunch of random stuff as well too see if anything else breaks :)

UkrainianBanderasCat committed:
Fixed not clearing text area

UkrainianBanderasCat committed:
Merge branch 'main' into notes

UkrainianBanderasCat committed:
Fixed problem with opening textarea

UkrainianBanderasCat committed:
Merge branch 'notes' of https://github.com/Lime-Game-Modding/Digital-Logic-Sim-Fan-Edit into notes

UkrainianBanderasCat committed:
Fixed undo system with notes

UkrainianBanderasCat committed:
Merge branch 'SebLague:main' into notes

UkrainianBanderasCat committed:
Fixed some problems

firecerne committed:
EEPROM 256x16

This is a chip that has similar capabilities to the ROM, + availability to be rewrote with its pins.
It saves.

logic-mindful committed:
Updated README

logic-mindful committed:
Fixed a grammar mistake

firecerne committed:
Made EEPROM rising-edge

Introduced CLOCK pin so that EEPROM chip is now rising-edge. It also reduces performance issue of copying an array to another each simulation cycle which is what was needed previously.

logic-mindful committed:
Merge pull request #1 from logic-mindful/organizational-stuff

Updated README

firecerne committed:
Merge pull request #2 from firecerne/EEPROM

The EEPROM chip.

firecerne committed:
Addition of Squircle to SebVis

SebVis now natively supports squircles drawing.

firecerne committed:
TypeHelper.cs for Vector2 type conversion.

firecerne committed:
Clickable Displays

firecerne committed:
Merge branch 'clickabledisplays' into buttons

firecerne committed:
Buttons and interaction finish

firecerne committed:
Remove unnecessary call

firecerne committed:
Adds Button to I/O collection

Oops ! Forgot it before !

logic-mindful committed:
Merge pull request #3 from firecerne/buttons

Added Buttons

firecerne committed:
Dispwitch (saves defaults)

firecerne committed:
Change appearance of the dipswitch.

logic-mindful committed:
Merge pull request #4 from firecerne/dipswitches

Dipswitches

firecerne committed:
Ready Build

firecerne committed:
ROM fix

didn't properly test earlier

logic-mindful committed:
Merge pull request #5 from firecerne/dipswitches

Change settings for build

firecerne committed:
Fix Dipswitch

Yay !

firecerne committed:
Merge pull request #6 from firecerne/dipswitches

ROM fix

BeboKhouja committed:
Add RTC

BeboKhouja committed:
Initial commit for SPS chip

BeboKhouja committed:
Fix SPS count

firecerne committed:
Ram builtin chip + secret "All Black Builtin"

RAM builtin chip. Works exactly like you would expect. Initial state of RAM is random.

You can make all of the built in chips all black by making the name of the projcet include the string "ahic"

firecerne committed:
Constant Chip

firecerne committed:
Merge pull request #7 from firecerne/ram

Ram builtin chip + secret "All Black Builtin"

firecerne committed:
Merge branch 'main' into constant

logic-mindful committed:
Merge pull request #8 from firecerne/constant

Constant Chip

Dom-c-d committed:
Initial Commit:
Custom pin location on subchips - snaps to any edge.
Pins now have a LocalOffset & face (0-3) to indicate which way to face and the offset along the corrosponding edge. 0 = top moves clockwise. There is still reference to LocalPosY on pins but this is just handled as the offset

Indicators to point in/out to indicate what pin type each is as they can now be on any edge
added in settings wheel for various display options for these indicators - Tab to toggle is bound to the existing tab detection for pin names

ChipCustomizationMenu now has a wheel to toggle Default Layout and custom layout. Will automatically toggle to custom on dragging pin. - changing back resets the layout.

TODO:
-Fix resizing, currently if you enlarge a chip, move the pin to the edge and then shrink the pin does not move to stay in bounds.

-Need to change MinSize Calculations to account for pin movement and now the size.X needs to check for pins along top and bottom at time of resizing.

-Load Custom layout on the chip customisation menu

- Potential improvement to single pin output shape/angle.
-slightly alter size of text on settings wheel

Dom-c-d committed:
Min Chip size is nolonger readonly. Now have a function that will Recalculate Min Size during chip resizing to account for any relocated pins. X now checks name min requirement against Pin min requirement to determine min X

If a chip is on the edge when resizing it will move the chip in with it, however this can push the edge chip(s) under other chips. Need to add in "pushing" behaviour until the min size is reached and all are spaced as close as possible

Dom-c-d committed:
Rewrote customLayout to to fix issue from last commit, now clamps pins onto chip and push pins inwards to ensure no overlap. sweeps both directions to ensure both ends stay on chip and move pins as needed.

Dom-c-d committed:
Rewrote customLayout to to fix issue from last commit, now clamps pins onto chip and push pins inwards to ensure no overlap. sweeps both directions to ensure both ends stay on chip and move pins as needed.

Dom-c-d committed:
Fixed gitIgnore

Dom-c-d committed:
Pulled latest Community edit and resolved conflicts in project description save

Dom-c-d committed:
Customisation menu now loads existing custom layouts if editing a previously created chip.
Fixed red highlight for overlapping pins not drawing horizontally.
Fixed issue of not calculating and setting chip min size when going from custom to default layout via wheel.

Dom-c-d committed:
Missed a typo which was causing some issues. Now working (I hope)

firecerne committed:
Support for pins of any size

Max limit of 16 bits for usable pins.

firecerne committed:
Loading fix

firecerne committed:
Merge branch 'main' into feat-RTC

firecerne committed:
Merge pull request #10 from BeboKhouja/feat-RTC

Add RTC

firecerne committed:
Improved Save Compatibility

firecerne committed:
Reorganized for ease of use.

Dom-c-d committed:
Merge branch 'main' into MovingPins

BeboKhouja committed:
Remove spct

BeboKhouja committed:
Delete Lua leftovers from unpublished branch

BeboKhouja committed:
Update gitignore to ignore vscode dirs

BeboKhouja committed:
Merge branch 'main' into feat-SPSChip

BeboKhouja committed:
Remove vscode dirs

BeboKhouja committed:
Gitignore projectversion

BeboKhouja committed:
Merge branch 'main' into feat-SPSChip

BeboKhouja committed:
Change projectversion

logic-mindful committed:
Update README.md

logic-mindful committed:
Merge branch 'main' into organizational-stuff

logic-mindful committed:
Merge pull request #12 from logic-mindful/organizational-stuff

Link to Task Management

BeboKhouja committed:
Update README.md

logic-mindful committed:
Merge pull request #13 from BeboKhouja/patch-1

Correct typos on README

logic-mindful committed:
Merge branch 'main' into feat-SPSChip

logic-mindful committed:
Merge pull request #11 from BeboKhouja/feat-SPSChip

Add SPS chip

logic-mindful committed:
Detector Gate, corrected ProjectVersion, code formatting

logic-mindful committed:
Create codeql.yml

logic-mindful committed:
Update codeql.yml

logic-mindful committed:
Update codeql.yml

logic-mindful committed:
Merge pull request #15 from logic-mindful/advanced-code-scanning

advanced-code-scanning

logic-mindful committed:
Update Simulator.cs

logic-mindful committed:
Merge branch 'main' into detector

BeboKhouja committed:
Initial commit for stats

BeboKhouja committed:
Add collection stats

logic-mindful committed:
Merge pull request #14 from logic-mindful/detector

Detector Gate, corrected ProjectVersion, code formatting

BeboKhouja committed:
Add num of chips in a chip

firecerne committed:
No max pincount

Slightly more performant : will need improvements.
(8% less performant, but could actually become more performant I think)

Still need to add custom Merge/Split, Buses, and the menu for those.

BeboKhouja committed:
Fix close button

BeboKhouja committed:
Add proj stats

firecerne committed:
Pin conflicts

The new implementation now shares the same features as the old one, simulation speaking.

firecerne committed:
Debug.Log cleanup

Cleaned a few debug logs.

BeboKhouja committed:
Add total chips used

firecerne committed:
Adding bus types to project description

firecerne committed:
Procedural generation of Splits

yeah pretty much the title

BeboKhouja committed:
Make close button drift less

BeboKhouja committed:
Make drifting not possible, instead the size can be under or over

logic-mindful committed:
Merge branch 'main' into feat-Stats

logic-mindful committed:
Merge pull request #16 from BeboKhouja/feat-Stats

Add stats feature

logic-mindful committed:
Update .gitignore

logic-mindful committed:
Merge pull request #18 from logic-mindful/logic-mindful-gitignore

Update .gitignore

firecerne committed:
Phew ! Merge + optimization!

firecerne committed:
BitArrayHelper cleanup

firecerne committed:
Order fix for compatibility between versions

firecerne committed:
Default chip collections for default pin/merge/split/bus

firecerne committed:
Menu to add special Pins/MergeSplit/Bus

firecerne committed:
Add keyboard shortcut

firecerne committed:
delete some useless test data

firecerne committed:
Update .gitignore

firecerne committed:
Deleting more useless Testdata

firecerne committed:
Delete s.json

firecerne committed:
Bigger wires and pins use less space

firecerne committed:
Merge branch 'main' into variable-wire-sizes

firecerne committed:
Compat fixes I think

firecerne committed:
Removed unnecessary and build breaking imports in SpecialChipMakerMenu.cs

firecerne committed:
Merge pull request #17 from firecerne/variable-wire-sizes

Variable wire sizes support
Includes : 
 - menu to add different wire sizes (up to 65 536 bits but please don't?). Also automatically adds the associated BUS subcomponent.
 - menu to add different split/merge chips. Works in pair.
 - slight modifications to the rendering of wires so that they're another color when above 64 bits (do distinguish bigger wires better) and don't render all of the wires.
 - slight modifications to the rendering of the pins to display a colored band indicating the depth of the pin to distinguish between pins of seemingly the same size but of different depth.

logic-mindful commented:
Also suggested in #510

BeboKhouja committed:
Fix SPS flags

BeboKhouja committed:
Fix stats and clock down sps

firecerne committed:
Fix for display of 32bit pins and higher

BeboKhouja committed:
Update chips with 16 and 32 bit wires

logic-mindful committed:
Merge pull request #21 from firecerne/variable-wire-sizes

Fix for display of 32bit pins and higher

logic-mindful committed:
Merge branch 'main' into patch-2

firecerne committed:
Fixed name of generated Split/Merge

For compatibility issues with previously created projects

logic-mindful committed:
Merge pull request #22 from firecerne/variable-wire-sizes

Fixed name of generated Split/Merge

BeboKhouja committed:
Merge branch 'main' of https://github.com/logic-mindful/Digital-Logic-Sim-Community-Edit into patch-2

BeboKhouja committed:
Do not remove old merge/splits

BeboKhouja committed:
Fix proj description

BeboKhouja committed:
Fix modded version

logic-mindful committed:
Merge pull request #20 from BeboKhouja/patch-2

Fix bugs with my chips and add new bit counts

firecerne committed:
Merge branch 'main' into MovingPins

firecerne committed:
Merge pull request #9 from Dom-c-d/MovingPins

Custom pin layouts on Subchips

firecerne committed:
Adaptations for higher bitcount + snap

logic-mindful committed:
Merge pull request #23 from firecerne/custom-layout-fixes

Adaptations of Dom's custom pin layouts for higher bitcount + snap

firecerne committed:
Quick bugfix, also removes clutter

firecerne committed:
Reassign base sizes to ROM and EEPROM

otherwise it breaks preexisting projects bruh

firecerne committed:
Fixing the fix

Forgot to fix EEPROM write

firecerne committed:
Makes the special chip maker more reassuring

Confirms when a pin size is added
Adds the option to close without saving just in case.

firecerne committed:
Special chip maker saving fix

Now, closing without saving actually closes without saving.
Saving only happens when saving.

firecerne committed:
Pressing escape closes the special chip maker menu

Pretty nifty huh ?

firecerne committed:
Rom Fix

firecerne committed:
M/S maker fixed (was a typo)

firecerne committed:
Removes unused code + fixes i/o alignment

firecerne committed:
Fix RAM performance

firecerne committed:
Merge pull request #24 from firecerne/main

Quick bugfix, also removes clutter

UkrainianBanderasCat committed:
Merge branch 'notes' into notes
